File: 151001/Shcherba/lab4/publicator/src/main/java/com/example/publicator/dto/TweetResponseTo.java
package com.example.publicator.dto;

import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TweetResponseTo {
    private int id;
    private int userId;
    @Size(min = 2, max = 64)
    private String title;
    @Size(min = 4, max = 2048)
    private String content;
    private Timestamp created;
    private Timestamp modified;
    private int markerId;
}

File: 151004/Golovchuk/Lab2/DC_Lab1/DTO/TweetResponseTo.cs
using DC_Lab1.DTO.Interface;
using DC_Lab1.Models;

namespace DC_Lab1.DTO
{
    public class TweetResponseTo(int Id, int? creatorId, string? Title, string? Content, string? Created, string? Modified ,ICollection<Post> Posts, Creator? Creator) : IResponseTo
    {
        public int Id { get; set; } = Id;

        public int? creatorId { get; set; } = creatorId;

        public string? Title { get; set; } = Title;

        public string? Content { get; set; } = Content;

        public string? Created { get; set; } = Created;

        public string? Modified { get; set; } = Modified;

        public virtual ICollection<Post> Posts { get; set; } = Posts;

        public virtual Creator? Creator { get; set; } = Creator;
    }
}
